opm/simulators/flow/FlowMainEbos.hpp

```diff
@@ -163,7 +163,7 @@ namespace Opm
             EWOMS_END_PARAM_REGISTRATION(TypeTag);
 
             // read in the command line parameters
-            int status = Ewoms::setupParameters_<TypeTag>(argc, const_cast<const char**>(argv), /*doRegistration=*/false);
+            int status = Ewoms::setupParameters_<TypeTag>(argc, const_cast<const char**>(argv), false, true, true);
```

Comment:
Why do we allow unused parameters? Shouldn't this PR (if it is correct) make #1845 abosolete?

BTW: without your helpful comments about what bool is what, this is rather complicated and unreadable. May I suggest adding `/*doRegistration*/` and the likes? That would really help.

How often are we actually calling setupParameters_ in flow?

Why the trailing underline in the name? Wasn't that used to mark functions considered internal? Yet. this one is used externally. That seems a bit confusing.

Reply:
> Why do we allow unused parameters?

at this point we cannot be sure which parameters are available: extensions like polymer, solvent and thermal might introduce some of their own.

> Shouldn't this PR (if it is correct) make #1845 abosolete?

no, not currently.

> BTW: without your helpful comments about what bool is what, this is rather complicated and unreadable. May I suggest adding /*doRegistration*/ and the likes? That would really help.

almost all places in the opm-simulators (except `ebos/`), opm-common and and opm-grid written by myself do not use this. I thought I adapt my coding style to the surroundings...

> How often are we actually calling setupParameters_ in flow?

`flow`: twice
`mebos`: thrice
`ebos*`: once

> Why the trailing underline in the name? Wasn't that used to mark functions considered internal? Yet. this one is used externally. That seems a bit confusing.

it is supposed to be an internal function only used by `Ewoms::start()`. since `flow` and `mebos` need to do some low-level meddling, they need to use it as well...